api/src/main/java/org/openmrs/module/patientflags/api/impl/FlagServiceImpl.java

```diff
@@ -688,10 +707,8 @@ private List<Flag> getFlags(List<PatientFlag> patientFlags) {
 
 	@Override
 	public Future<?> evaluateAllFlags() {
-		if (executor == null) {
-			executor = Executors.newSingleThreadExecutor();
-		}
-		
+		executor = Executors.newSingleThreadExecutor();

```

Comment:
[in this line](https://github.com/openmrs/openmrs-module-patientflags/blob/a1e4aa6cbb543369c66b6395422136ab292eec48/api/src/main/java/org/openmrs/module/patientflags/api/impl/FlagServiceImpl.java#L106) , I initialized the executor as a thread pool. If the `generateFlagForPatient` method is called first, the executor is already initialized as thread pool, so then `evaluateAllFlags` uses a thread pool instead of a single thread.  after removing the null check when `evaluateAllFLag` method calls it initialize the executor as a single thread.